### ISSUE TITLE
drake workflow: offloading to functions

### ISSUE THREAD

**Issue by wlandau** [sol-eng/tensorflow-w-r#5]
@edgararuiz, I took a more serious look at your notebook, and I think it would be nice to offload code from the plan to user-defined functions. I just created a [new official `drake` example](https://github.com/wlandau/drake-examples/tree/master/tensorflow) based on your work. The plan is smaller.

```r
plan <- drake_plan(
  data = read_csv(file_in("data/customer_churn.csv"), col_types = cols()) %>%
    initial_split(prop = 0.3),
  training_recipe = customer_churn_recipe(data),
  fitted_model = fit_churn_model(training_recipe, data),
  plotted_model = plot(fitted_model),
  confusion_matrix = get_confusion_matrix(training_recipe, data),
  report_step = rmarkdown::render(
    knitr_in("results.Rmd"),
    output_file = file_out("results.html"),
    quiet = TRUE
  )
)
```

![graph](https://user-images.githubusercontent.com/1580860/54773514-1fd01980-4be0-11e9-889b-6e656e3fad1b.png)

That way, we duplicate the data less often, and the plan can concentrate on declaring multiple models and multiple kinds of summaries for users who want to scale up. Plus, compiled models seem to no longer function after they are loaded from `drake` 's default on-disk cache, so I do not think we can keep the compiled model as a target.

Feedback is welcome. I feel like I have been learning a lot about recipes in the process.

**Comment by wlandau** [comment]
By the way, I do plan to submit a PR. I just need to finish some internal company stuff to get permission first.

**Comment by edgararuiz-zz** [comment]
Fixed by 7e0a686755188ead872802a19d2d9074a7019e47
